api/views.py

```diff
@@ -83,12 +92,17 @@ def post(self, request, format=None):
         return response.Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
 
-class Matches(generics.GenericAPIView):
+class Matches(viewsets.ModelViewSet):
     """
     View for requesting new matches for a user.
     """
 
-    serializer_class = UserSerializer
+    def get_serializer_class(self):
+        if self.action == "list":
+            return ProfileSerializer
+        if self.action == "partial_update":
+            return MatchStatusSerializer
+        return super().get_serializer_class()
```

Comment:
To me, this is a signal to use two separate endpoints and views, but it's not a big deal either way.

Reply:
I don't really like it much either. If I move the matches from the matchstatus model to many-to-many on the profile, then I could probably avoid this since I'd be able to use the same profile serializer for updating and getting matches.

I spent a good amount of time thinking about how to do this best. I don't think there should be two endpoints because I view the /api/matches/ endpoint as referring to a collection of matches that can be operated on. So naturally, getting matches and updating matches should both be operating on the matches collection, regardless of how the collection is managed on the backend.